lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java

```diff
@@ -84,4 +87,11 @@ private static DumperOptions getDumperOptions() {
     public void close() throws IOException {
         writer.close();
     }
+
+    private static class LockDescriptorAwareRepresenter extends Representer {
+        LockDescriptorAwareRepresenter() {
+            super();
+            this.representers.put(LockDescriptor.class, data -> representScalar(Tag.STR, data.toString()));
```

Comment:
Just `getLockIdAsString` doesn't do ASCII encoding. `toString` actually returns more readable output, whenever possible.

The idea to keep pointer, instead of String as long as possible is for less memory footprint. SnakeYaml is smart enough not to serialize the whole output first before dumping, so I let it decide when it needs an actual String.